controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -354,6 +359,51 @@ private HttpResponse tenant(Tenant tenant, HttpRequest request) {
         return new SlimeJsonResponse(slime);
     }
 
+    private HttpResponse tenantCost(String tenantName, HttpRequest request) {
+        return controller.tenants().get(TenantName.from(tenantName))
+                .map(tenant -> tenantCost(tenant, request))
+                .orElseGet(() -> ErrorResponse.notFoundError("Tenant '" + tenantName + "' does not exist"));
+    }
+
+    private HttpResponse tenantCost(Tenant tenant, HttpRequest request) {
+        var slime = new Slime();
+        var objectCursor = slime.setObject();
+        var monthsCursor = objectCursor.setArray("months");
+
+        return new SlimeJsonResponse(slime);
+    }
+
+    private HttpResponse tenantCost(String tenantName, String monthString, HttpRequest request) {
+        Optional<Tenant> tenant = controller.tenants().get(TenantName.from(tenantName));
+        Optional<LocalDate> month = tenantCostParseMonth(monthString);
+
+        if (tenant.isEmpty()){
+            return ErrorResponse.notFoundError("Tenant '" + tenantName + "' does not exist");
+        }
+
+        if (month.isEmpty()) {
+            return ErrorResponse.badRequest("Could not parse month '" + monthString + "'");
+        }
+
+        return tenantCost(tenant.get(), month.get(), request);
+    }
+
+    private Optional<LocalDate> tenantCostParseMonth(String monthString) {
+        var formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+        try {
+            return Optional.of(LocalDate.parse(monthString, formatter).withDayOfMonth(1));
+        } catch (DateTimeParseException ignored) {
+            return Optional.empty();
```

Comment:
It's better to just rethrow this as an `IllegalArgumentExeption` and remove the explicit `month.isEmpty()` check, the `handle` method will then convert this into a bad request response.